XenModel/Actions/Network/CreateSriovAction.cs

```diff
@@ -81,19 +80,41 @@ protected override void Run()
             int inc = 100 / (selectedPifs.Count + 1);
             int lo = inc;
 
-            foreach (PIF thePif in selectedPifs)
+            // Create the new network            
+            XenRef<XenAPI.Network> networkRef = XenAPI.Network.create(Session, newNetwork);
+
+            try
+            {
+                foreach (PIF thePif in selectedPifs)
+                {
+                    RelatedTask = Network_sriov.async_create(Session, thePif.opaque_ref, networkRef);
+                    PollToCompletion(lo, lo + inc);
+                    lo += inc;
+                }
+            }
+            catch(Exception)
             {
-                RelatedTask = Network_sriov.async_create(Session, thePif.opaque_ref, networkRef);
-                PollToCompletion(lo, lo + inc);
-                lo += inc;
+                DestroyNetwork(networkRef);
+                throw;

```

Comment:
How about the `Network_sriov` objects that we successfully created in the `foreach (PIF thePif in selectedPifs)` loop above?
Do we need to destroy those as well if we destroy the network?

Reply:
Thank you for reminding. I think we don't need to destroy the created ```Network_sriov``` objects. There should be two situation. The first one is that when some NICs have been enabled SR-IOV and the  ```Network_sriov``` objects have been created. So we also shouldn't destroy the network and the  ```Network_sriov``` objects. The second one is that no NIC has been enabled SR-IOV, so we need to destroy the network and there is no  ```Network_sriov``` object need to be deleted.

Reply:
Yes, but what happens if you have say 3 NICs with enabled SR-IOV, you create 2 Network_sriov objects successfully and the third creation fails for some unforseen reason? You destroy the network, but then you have the first 2 Network_sriov objects left without one - should those 2 be deleted too?

Reply:
Thanks. I think the problem you mentioned has been solved in the latest commits. The ```if(lo == inc)``` was used for this. Only when the first time the creation fails, the network will be destroyed.

Reply:
Hi @kc284 and @MihaelaStoica, I have already updated the PR. Please correct me if you have further comments. Thank you very much.

Reply:
Ah ok, I hadn't seen the second commit when I wrote the comment.
One more question with regards to the polling, and I apologise if I'm missing something, but why are you using ` int inc = 100 / (selectedPifs.Count + 1); int lo = inc;` instead of ` int inc = 100 / selectedPifs.Count; int lo = 0;` (and checking obviously that Count is not 0)? With the first method the progress reported gives the wrong impression to the user.

Reply:
Also, could you please squash the commits. Thanks.

Reply:
Thank you. I will refine it.